frontend/src/widgets/CoursesAssociatedWithActivityReports.js

```diff
@@ -0,0 +1,230 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { DECIMAL_BASE } from '@ttahub/common';
+import HorizontalTableWidget from './HorizontalTableWidget';
+import WidgetContainer from '../components/WidgetContainer';
+import useSessionSort from '../hooks/useSessionSort';
+import { COURSES_PER_PAGE } from '../Constants';
+
+export const parseValue = (value) => {
+  const noCommasValue = value.replaceAll(',', '');
+  const parsedValue = parseInt(noCommasValue, DECIMAL_BASE);
+  if (Number.isNaN(parsedValue)) {
+    return value;
+  }
+  return parsedValue;
+};
+function CoursesAssociatedWithActivityReports({
+  data,
+  loading,
+  resetPagination,
+  setResetPagination,
+  perPageNumber,
+}) {
+  const [courseUse, setCourseUse] = useState([]);
+  const [courseCount, setCourseCount] = useState(0);
+  const [localLoading, setLocalLoading] = useState(false);
+  const [perPage] = useState(perPageNumber);
```

Comment:
if this never changes, why do we put it in use state? why not just use the prop passed in?

Reply:
Correct me if I'm wrong but I think some of this does change in the component for things like sorting.

Reply:
It also is handling page change in this level.

Reply:
When does perPage change? It can't; there's no second half of the useState: [perPage, setPerPage]. It will always be the initial value of perPageNumber that was passed in.

Reply:
ah did you mean just perPage its difficult to see what your where referring to in this comment.

Reply:
gotcha should have said perPage